README.md

```diff
@@ -1,48 +1,63 @@
 
-gProfiler can upload the result to the [Granulate Performance Studio](https://profiler.granulate.io/), which aggregates the results from different instances over different periods of time and can give you a holistic view of what is happening on your entire cluster.
+gProfiler can upload its results to the [Granulate Performance Studio](https://profiler.granulate.io/), which aggregates the results from different instances over different periods of time and can give you a holistic view of what is happening on your entire cluster.
 To upload results, you will have to register and generate a token on the website.
 
-## Requirements
-gProfiler works on **Linux** and requires **Python 3.6+** to run.
+gProfiler runs on Linux.
 
-The `nsenter` program needs to be installed for Java profiling. For Debian/Ubuntu, install the `util-linux` package.
+# Running
 
-It can produce specialized stack traces for the following runtimes:
-* Java runtimes (version 7+) based on the HotSpot JVM,
-including the Oracle JDK and other builds of OpenJDK like AdoptOpenJDK and Azul Zulu.
-* The CPython interpreter, versions 2.7 and 3.5-3.9.
-  * eBPF profiling requires Linux 4.14 or higher.
+This section describes the possible options to control gProfiler's output, and the various execution modes (as a container, as an executable, etc...)
 
-gProfiler can profile Python applications with low overhead using eBPF. This requires kernel
-headers to be installed.
+## Output options
 
-## Running from source
-```bash
-pip3 install -r requirements.txt
-./scripts/build.sh
-```
+gProfiler can produce output in two ways:
 
-### Usage
-Run the following **as root**:
-```bash
-python3 -m gprofiler [options]
-```
+* Create an aggregated, collapsed stack samples file (`profile_<timestamp>.col`)
+  and a flamegraph file (`profile_<timestamp>.html`).
 
-## Running as a docker container
-Run the following to have gprofiler running continuously, uploading to Granulate Performance Studio:
+  Use the `--output-dir`/`-o` option to specify the output directory.
+
+* Send the results to the Granulate Performance Studio for viewing online with
+  filtering, insights, and more.
+
+  Use the `--upload-results`/`-u` flag. Pass the `--token` option to specify the token
+  provided by Granulate Performance Studio, and the `--service-name` option to specify an identifier
+  for the collected profiles, as will be viewed in the [Granulate Performance Studio](https://profiler.granulate.io/). *Profiles sent from numerous
+  gProfilers using the same service name will be aggregated together.*
+
+Note: both flags can be used simultaneously, in which case gProfiler will create the local files *and* upload
+the results.
+
+## Profiling options
+* `--profiling-frequency`: The sampling frequency of the profiling, in *hertz*.
+* `--profiling-duration`: The duration of the each profiling session, in *seconds*.
+* `--profiling-interval`: The interval between each profiling session, in *minutes*.
```

Comment:
I would change this argument (in the code as well) to reflect the interval between 2 consecutive runs and not between the start of both of them.
such that it would not be legit to run overlapping profilers.

Reply:
It's not entirely accurate. The wait is `max(interval * 60 - time_spent, 0)` so it does take into account the time we've spent in the session. I'll clarify.

Reply:
yeah - I understand but changing the argument to what I suggested disallows the option to specify an incorrect value

Reply:
Agree that interval can be checked to be greater than duration (otherwise it makes less sense)

Reply:
I'll change in another PR.